src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,278 @@
-    super.onActivityStopped(activity);
+  @Override
+  public void screen(ScreenPayload screen) {
+    super.screen(screen);
 
+    logger.verbose(
+            "The Firebase SDK gathers screen calls automatically. Segment does map manual screen " +
+                    "methods to Firebase.");
   }
 
-  @Override public void identify(IdentifyPayload identify) {
-    super.identify(identify);
+  @Override
+  public void track(TrackPayload track) {
+    super.track(track);
 
-    if (!isNullOrEmpty(identify.userId())) {
-      mFirebaseAnalytics.setUserId(identify.userId());
+    String event = track.event();
+    String eventName = mapEvent(event);
+    Properties properties = track.properties();
+
+    try {
+      Bundle formattedProperties = formatProperties(properties);
+      firebaseAnalytics.logEvent(eventName, formattedProperties);
+      logger.verbose("firebaseAnalytics.logEvent(%s, %s);", eventName, formattedProperties);
+    } catch (Exception e) {
+      logger.verbose(
+              "You must set `currency` in your event's property object to accurately "
+                      + "pass 'value' to Firebase. This event was not sent to Firebase.");
     }
+  }
 
-    // mFirebaseAnalytics.setUserProperty(key, value);
+  private String mapEvent(String event) {
+    String eventName = event;
+    if (eventMapper.containsKey(eventName)) {
+      eventName = eventMapper.get(eventName);
+    }
+    if (eventName.contains(" ")) {
+      eventName = eventName.trim().toLowerCase().replaceAll(" ", "_");
+    }
+    if (eventName.length() > 40) {
+      eventName = trimKey(eventName);
+    }
+    return eventName;
+  }
 
+  private Bundle formatProperties(Properties properties) throws Exception {
+    if ((properties.revenue() != 0 || properties.total() != 0) && isNullOrEmpty(properties.currency())) {
+      throw new Exception();
+    }
+    Bundle bundle = new Bundle();
+    for (Map.Entry<String, Object> entry : properties.entrySet()) {
+      String property = entry.getKey();
+      property  = mapProperty(property);
+      if (entry.getValue() instanceof Integer) {
+        int value = (int) entry.getValue();
+        bundle.putInt(property, value);
+        logger.verbose("bundle.putInt(%s, %s);", property, value);
+        continue;
+      }
+      if (entry.getValue() instanceof Double) {
+        double value = (double) entry.getValue();
+        bundle.putDouble(property, value);
+        logger.verbose("bundle.putDouble(%s, %s);", property, value);
+        continue;
+      }
+      if (entry.getValue() instanceof Long) {
+        long value = (long) entry.getValue();
+        bundle.putLong(property, value);
+        logger.verbose("bundle.putLong(%s, %s);", property, value);
+        continue;
+      }
+      if (entry.getValue() instanceof String) {
+        String value = String.valueOf(entry.getValue());
+        if (ISO_DATE.matcher(value).matches()) {
+          value = value.substring(0, 10);
```

Comment:
This will probably go away as pointed out in https://github.com/segment-integrations/analytics-android-integration-firebase/pull/1/files#r137365797.

If not, why is substring(0, 10) being used here?

Reply:
I understand why this is being done.